modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java

```diff
@@ -153,9 +142,11 @@ public CompletableFuture<Index<?>> createIndexAsync(
         try {
             validateName(indexName);
 
-            CompletableFuture<Index<?>> future = new CompletableFuture<>();
+            var canonicalIndexName = parseCanonicalName(canonicalName(schemaName, indexName));
+
+            CompletableFuture<Boolean> future = new CompletableFuture<>();
 
-            var canonicalName = canonicalName(schemaName, tableName);
+            var canonicalName = parseCanonicalName(canonicalName(schemaName, tableName));
```

Comment:
TableManager already has such a normalization what wrong here ?

Reply:
I wouldn't use TableManager as a reference here, because it kinda mess of public and internal API. This `normalisation` should be done in public facade only

Reply:
All such activity is planning into https://issues.apache.org/jira/browse/IGNITE-17677 or you want this concrete case to be fixed ?